.github/workflows/run_colossalqa_unit_tests.yml

```diff
@@ -0,0 +1,52 @@
+name: Run colossalqa unit tests
+
+on:
+  pull_request:
+    types: [synchronize, opened, reopened]
+    paths:
+      - 'applications/langchain/colossalqa/**'
+      - 'applications/langchain/requirements.txt'
+      - 'applications/langchain/setup.py'
+      - 'applications/langchain/tests/**'
+      - 'applications/langchain/pytest.ini'
+
+jobs:
+  tests:
+    name: Run colossalqa unit tests
+    if: |
+      github.event.pull_request.draft == false &&
+      github.base_ref == 'main' &&
+      github.event.pull_request.base.repo.full_name == 'hpcaitech/ColossalAI'
+    runs-on: [self-hosted, 8-gpu]
+    container:
+      image: hpcaitech/pytorch-cuda:1.12.0-11.3.0
+      volumes:
+        - /data/scratch/test_data_colossalqa:/data/scratch/test_data_colossalqa
+        - /data3/data:/data3/data
+      options: --gpus all --rm
+    timeout-minutes: 30
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: Checkout ColossalAI
```

Comment:
Please specify the specific version of `colossalai`.

Just in case some APIs are changed in the future.